chainio/clients/wallet/fireblocks_wallet.go

```diff
@@ -18,12 +19,20 @@ import (
 var _ Wallet = (*fireblocksWallet)(nil)
 
 type fireblocksWallet struct {
+	mu sync.Mutex
```

Comment:
what is this mutex for? to protect concurrent writes to nonceToTxID and txIDToNonce? Can we name the mutex more explicitly?
Also write a small explanation as to why its needed. Is the idea that I could be trying to send multiple txs from different threads? Unclear to me why this would happen.

Reply:
oh just realized there's also concurrent access between send and gettxreceipt

Reply:
yup I'll add comment